src/Microsoft.VisualBasic/tests/OperatorsTests.cs

```diff
@@ -5456,11 +5456,11 @@ private static void AssertEqual(object expected, object actual)
         {
             if (expected is double expectedDouble && actual is double actualDouble)
             {
-                Assert.Equal(expected.ToString(), actual.ToString());
+                Assert.Equal(expectedDouble, actualDouble, 10);
             }
             else  if (expected is float expectedFloat && actual is float actualFloat)
             {
-                Assert.Equal(expected.ToString(), actual.ToString());
+                Assert.Equal(expectedFloat, actualFloat, 10);
```

Comment:
I don't think xUnit has an `Assert.Equal` overload that takes a `float`, so this will be converting to double and comparing those.

`10` digits is also more than needed. `System.Single` has approx. 6-9 digits of precision, and you need at most 9 to uniquely represent every value.